execution_engine_testing/tests/src/test/regression/step_regression.rs

```diff
@@ -0,0 +1,196 @@
+
+    let bids_after_slashing: Bids = builder.get_bids();
+    let account_1_bid = bids_after_slashing.get(&ACCOUNT_1_PK).unwrap();
+    assert!(account_1_bid.inactive());
+    assert!(account_1_bid.staked_amount().is_zero());
+
+    let bids_after_slashing: Bids = builder.get_bids();
+    assert_ne!(
+        bids_before_slashing, bids_after_slashing,
+        "bids table should be different before and after slashing"
+    );
+
+    // seigniorage snapshot should have changed after auction
+    let after_auction_seigniorage: SeigniorageRecipientsSnapshot =
+        builder.get_value(auction_hash, SEIGNIORAGE_RECIPIENTS_SNAPSHOT_KEY);
+    assert!(
+        !before_auction_seigniorage
+            .keys()
+            .all(|key| after_auction_seigniorage.contains_key(key)),
+        "run auction should have changed seigniorage keys"
+    );
+
+    let step_request_2 = StepRequestBuilder::new()
+        .with_parent_state_hash(builder.get_post_state_hash())
+        .with_protocol_version(ProtocolVersion::V1_0_0)
+        .with_slash_item(SlashItem::new(ACCOUNT_1_PK.clone()))
+        .with_reward_item(RewardItem::new(ACCOUNT_1_PK.clone(), BLOCK_REWARD / 2))
+        .with_reward_item(RewardItem::new(ACCOUNT_2_PK.clone(), BLOCK_REWARD / 2))
+        .with_next_era_id(EraId::from(2))
+        .with_era_end_timestamp_millis(eras_end_timestamp_millis_2.as_millis().try_into().unwrap())
+        .build();
+
+    let StepSuccess {
+        execution_effect: transforms_2,
+        ..
+    } = builder.step(step_request_2).expect("should execute step");
+
+    let cl_u512_zero = CLValue::from_t(U512::zero()).unwrap();
+
+    let balances_1: BTreeSet<&Key> = transforms_1
+        .transforms
+        .iter()
+        .filter_map(|(key, transform)| match transform {
+            Transform::Write(StoredValue::CLValue(cl_value))
+                if key.as_balance().is_some() && cl_value == &cl_u512_zero =>
+            {
+                Some(key)
+            }
+            _ => None,
+        })
+        .collect();
+
+    assert_eq!(
+        balances_1.len(),
+        4,
+        "distribute should create 4 purses and zero out their balance"
+    );
+
+    let balances_2: BTreeSet<&Key> = transforms_2
+        .transforms
+        .iter()
+        .filter_map(|(key, transform)| match transform {
+            Transform::Write(StoredValue::CLValue(cl_value))
+                if key.as_balance().is_some() && cl_value == &cl_u512_zero =>
+            {
+                Some(key)
+            }
+            _ => None,
+        })
+        .collect();
+
+    assert_eq!(
+        balances_2.len(),
+        4,
+        "distribute should create 4 purses and zero out their balance"
+    );
+
+    let common_keys: BTreeSet<_> = balances_1.intersection(&balances_2).collect();
+    assert_eq!(common_keys.len(), 0, "there should be no commmon Key::Balance keys with Transfer::Write(0) in two distinct step requests");
+}
```

Comment:
I haven't read the whole test but I'm guessing that what we're looking for is a list of execution effects in each of the step requests and then compare that they don't overlap. If that's true, then we should be able to extract this code to a single method `fn run_step(...) -> ExecutionEffects`, call it twice and compare the keys that were written.